extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -497,11 +497,8 @@ bool TailFile::recoverState(const std::shared_ptr<core::ProcessContext>& context
 bool TailFile::getStateFromStateManager(std::map<std::string, TailState> &new_tail_states) const {
   std::unordered_map<std::string, std::string> state_map;
   if (state_manager_->get(state_map)) {
-    for (size_t i = 0U;; i++) {
-      std::string name;
-      try {
-        name = state_map.at("file." + std::to_string(i) + ".name");
-      } catch (...) {
+    for (size_t i = 0U;; ++i) {
+      if (state_map.find("file." + std::to_string(i) + ".name") == state_map.end()) {
         break;
       }
```

Comment:
I am genuinely curious what was the reasoning behind this code, could we be missing something by rewriting it to `std::map<...>::find`?

Reply:
I don't think we would miss anything. `std::unordered_map<...>::at()` only throws if the requested element is not found, the new code is also checking that. My goal was to eliminate exceptions as they are confusing during debugging and should not be used for control flow. And especially eliminating `catch (...)` from wherever possible.